### ISSUE TITLE
Row/column height seems not to work

### ISSUE THREAD

**Issue by knownasilya** [rkusa/pdfjs#21]
I've tried setting the height on the row and the columns, but nothing seems to work. The data dictates the height.

```js
var doc = pdfjs.createDocument({
  font: fonts.openSansRegular
});
var table = doc.table({
  width: 595.296,
  height: 841.896,
  widths: ['33.3%', '33.3%', '33.3%']
});
var row = table.tr();

data.forEach(function (item, index) {
  row.td(item.ownerName + '\n' + item.address + '\n' + item.city + (item.zip ? ', ' + item.zip : ''), {
    height: 300
  });

  // Create new row after every third column (3 column setup)
  if ((index + 1) % 3  === 0) {
    row = table.tr();
  }
});
```

**Comment by rkusa** [comment]
Tables do not support fixed heights, yet :-/

**Comment by knownasilya** [comment]
@rkusa any way I can help to land that feature?

**Comment by rkusa** [comment]
@knownasilya Yes, tell me how you'd expect the fixed height to work. E.g. should the last row take the remaining space, or should the rows be distributed among the fixed height, and how should content bigger than the fixed height be handled?

**Comment by knownasilya** [comment]
cc @ismyrnow

**Comment by knownasilya** [comment]
@rkusa seems like if the table height is set, the last row should fill i the remaining height, otherwise just use the height set.

If content overflows it should probably be hidden, since you are setting a fixed height.

**Comment by rkusa** [comment]
I took a look at how HTML and in comparison Pages and Word handle these situations. All three distribute the height among their rows and the height cannot be smaller than the content.

@knownasilya May I ask what your use case for a fixed height table in a PDF is?

**Comment by knownasilya** [comment]
@rkusa generating mailing labels, which come in specific sizes, since they are pre-cut stickers.

**Comment by rkusa** [comment]
That is, you'll also need some kind of vertical alignment option, doesn't you?

**Comment by knownasilya** [comment]
Not sure, but we could just start an implementation and see what's needed from there. Make it iterative, or just good enough for 1.0 and gain use cases for 2.0

**Comment by rkusa** [comment]
I've added a `height` option for rows and cells. A `height` option for the whole table is still on my todo

**Comment by knownasilya** [comment]
Nice, I'll give it a try

**Comment by knownasilya** [comment]
So that works, but if the content goes outside of the height it breaks the heights..

**Comment by rkusa** [comment]
@schurb Yes, imitates the behaviour found in HTML, Word, Pages, ... for your use case there is probably the need of an CSS overflow equivalent. I'll have a look ...

**Comment by schurb** [comment]
@rkusa I think you mean @knownasilya :)

**Comment by rkusa** [comment]
Haha... note to myself: dont answer issues before the first coffee :coffee:

**Comment by schurb** [comment]
Jep, such a universal rule :coffee:

**Comment by knownasilya** [comment]
Haha, and here I go breaking it. :neckbeard:

**Comment by rkusa** [comment]
I've started to implement the `overflow: hidden` option. Example:

```
  var table = doc.table({
    borderWidth: 1,
    borderColor: 0xdddddd,
    padding: 10,
    widths: [200]
  })

  var tr = table.tr({ height: 220, overflow: 'hidden' })
  tr.td(fixtures.lorem.short)
```

It is not well tested, yet. However, if you don't mind, it would be nice if you could test it against your usecase.

**Comment by rkusa** [comment]
`2.0.0-alpha.1` has a feature regression here, both `height` and `overflow` have to be re-implemented

**Comment by knownasilya** [comment]
Was this fixed in 2.0 release? Or still for 2.1?

**Comment by rkusa** [comment]
Still for 2.1, sorry 😔

**Comment by knownasilya** [comment]
Wait so height doesn't work now as well? If so I might not be able to update to 2.0 yet, aww..

**Comment by rkusa** [comment]
Yes, height does not work as well. If 1.x works fine for you, I don't think you are missing out one something with 2.0, yet

**Comment by chadRoberge** [comment]
Looking to use this library to generate checks, will also need the fixed height, and looking forward to it.  There a time frame for 2.1?

**Comment by rkusa** [comment]
I've pushed a `minHeight` option today
@chadRoberge Since those are features I am not needing for my own use-case, now ETA, sorry

**Comment by knownasilya** [comment]
What's the status here? Does `minHeight` replace what `height` used to be? What about `overflow`?

**Comment by rkusa** [comment]
@knownasilya I still have it in the back of my mind, but no update yet, I am afraid.

**Comment by knownasilya** [comment]
Would love to get both of these back. Finally upgraded due to node versions, but now having issues due to both of these missing.
